vcf2fhir/fhir_helper.py

```diff
@@ -488,72 +486,32 @@ def generate_final_json(self):
+                    createOrderedDict(i[v_c_c]['coding'][0], l1)
 
             if ((i['id'].startswith('dv-')) or (i['id'].startswith('rs-'))):
                 for q, j in enumerate(i['component']):
-                    od_component_code_coding = OrderedDict()
-                    if j['code']['coding'][0]["system"]:
-                        temp = j['code']['coding'][0]["system"]
-                        od_component_code_coding["system"] = temp
-                    if j['code']['coding'][0]["code"]:
-                        temp = j['code']['coding'][0]["code"]
-                        od_component_code_coding["code"] = temp
-                    if j['code']['coding'][0]["display"]:
-                        temp = j['code']['coding'][0]["display"]
-                        od_component_code_coding["display"] = temp
-                    if od['contained'][k]['component'][q]['code']['coding'][0]:
-                        temp = od_component_code_coding
-                        s1 = 'contained'
-                        s2 = 'component'
-                        od[s1][k][s2][q]['code']['coding'][0] = temp
-
-                    od_componentvalue_codeable_concept = OrderedDict()
-                    if 'valueCodeableConcept' in j.keys():
-                        temp = j['valueCodeableConcept']['coding'][0]["system"]
-                        od_componentvalue_codeable_concept["system"] = temp
-                        if 'code' in j['valueCodeableConcept']['coding'][0]\
-                            .keys(
-                        ):
-                            t = j['valueCodeableConcept']['coding'][0]["code"]
-                            od_componentvalue_codeable_concept["code"] = t
-                        if 'display' in j['valueCodeableConcept']['coding'][0]\
-                            .keys(
-                        ):
-                            s1 = 'valueCodeableConcept'
-                            s2 = 'display'
-                            temp = j[s1]['coding'][0]["display"]
-                            od_componentvalue_codeable_concept[s2] = temp
-                        s1 = 'contained'
-                        s2 = 'component'
-                        s3 = 'valueCodeableConcept'
-                        temp = od_componentvalue_codeable_concept
-                        od[s1][k][s2][q][s3]['coding'][0] = temp
+                    od_contained_k_component_q = od_contained_k['component'][q]
+                    if od_contained_k_component_q['code']['coding'][0]:
+                        od_contained_k_component_q['code']['coding'][0] =\
+                            createOrderedDict(j['code']['coding'][0], l1)
+
+                    if v_c_c in j.keys():
+                        od_contained_k_component_q[v_c_c]['coding'][0] =\
+                            createOrderedDict(j[v_c_c]['coding'][0], l1)
 
             if (i['id'].startswith('rs-')):
                 od_RS = OrderedDict()

```

Comment:
Any reason why we are not changing this and similar others too.

Reply:
sorry, but I didn't understand. Are you talking about ' v_c_c '

Reply:
I mean order dict created at line 517 and below. Why are we not using this utility for it ?

Reply:
Ya, sorry for that unnecessary comment, I understood that when i saw the code part